pkg/hubble/metrics/api/api.go

```diff
@@ -120,6 +127,16 @@ func (h Handlers) ProcessFlow(ctx context.Context, flow *pb.Flow) error {
 	return processingErr
 }
 
+// ProcessPodDeletion queries all handlers for a list of MetricVec and removes
+// metrics directly associated to deleted pod.
+func (h Handlers) ProcessPodDeletion(pod *slim_corev1.Pod) {
+	for _, h := range h.handlers {
+		for _, mv := range h.ListMetricVec() {
+			h.Context().DeleteMetricsAssociatedWithPod(pod.GetName(), pod.GetNamespace(), mv)
```

Comment:
Should we do a `nil` check on `Context()` here? Otherwise we should document that `Context()` must not return nil.

Reply:
fixed